examples/platforms/simulation/ble.c

```diff
@@ -26,50 +26,196 @@
  *  POSSIBILITY OF SUCH DAMAGE.
  */
 
+#include "platform-simulation.h"
+
+#include <errno.h>
+
+#include <stdio.h>
+#include <stdlib.h>
 #include <openthread/platform/ble.h>
 
+#include "openthread/error.h"
+#include "utils/code_utils.h"
+
+#define PLAT_BLE_MSG_DATA_MAX 512
+static uint8_t sBleBuffer[PLAT_BLE_MSG_DATA_MAX];
+
+static int sFd = -1;
+
+static const uint16_t kPortBase = 10000;
+static uint16_t       sPort     = 0;
+struct sockaddr_in    sSockaddr;
+
+static void initFds(void)
```

Comment:
I recently added `simul_utils` (refactored similar code used by TREL and radio adding `utilsSocket` and a bunch of helper functions). It may be useful here as well?

Reply:
@abtink Can we agree that I will raise an issue for refactoring (and assign me to it)? From what I have seen most of the code except for the creation of the socked could be reused.

Reply:
Yes. SGTM